File: app/src/main/java/codes/wokstym/cookingrecipes/service/RecipeService.java
package codes.wokstym.cookingrecipes.service;

import java.util.List;
import java.util.UUID;

import codes.wokstym.cookingrecipes.models.IngredientDto;
import codes.wokstym.cookingrecipes.models.RecipeDto;
import codes.wokstym.cookingrecipes.models.ShoppingListDto;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface RecipeService {

    @GET("/ingredient/{id}")
    Call<IngredientDto> getIngredient(@Path("id") UUID id);

    @GET("/recipe/all")
    Call<List<RecipeDto>> getRecipes();

    @POST("/list/")
    Call<ShoppingListDto> getShoppingList(@Body List<UUID> uuidsOfRecipes);

}

File: app/src/main/java/codes/wokstym/cookingrecipes/service/RecipeService.kt
package codes.wokstym.cookingrecipes.service

import codes.wokstym.cookingrecipes.models.IngredientDto
import codes.wokstym.cookingrecipes.models.RecipeDto
import codes.wokstym.cookingrecipes.models.ShoppingListDto
import okhttp3.MultipartBody
import retrofit2.Call
import retrofit2.http.*
import java.util.*

interface RecipeService {

    @GET("/ingredient/{id}")
    fun getIngredient(@Path("id") id: UUID?): Call<IngredientDto>

    @GET("/recipe/all")
    fun getRecipes(): Call<List<RecipeDto>>

    @GET("/recipe/mealtimes")
    suspend fun getMealTimes(): List<String>

    @POST("/list/")
    fun getShoppingList(@Body uuidsOfRecipes: List<UUID>): Call<ShoppingListDto>

    @Multipart
    @POST("/recipe/")
    fun addRecipe(
        @Part("recipeDto") recipeDto: NewRecipeDto,
        @Part image: MultipartBody.Part?
    ): Call<RecipeDto>
}
